src/SmartSelect/Alignment.elm

```diff
@@ -1,46 +1,148 @@
-module SmartSelect.Alignment exposing (Alignment(..), containerClass, init, selectClass)
+module SmartSelect.Alignment exposing
+    ( Alignment
+    , init, getElements
+    , style
+    , params
+    , view
+    )
 
-import Browser.Dom exposing (Element)
+{-| Determine the Alignment for the select options container
+
+
+# Definitions
+
+@docs Alignment, Params
+
+
+# Init
+
+@docs init, getElements
+
+
+# Style
+
+@docs style
+
+
+# Test
+
+@docs params
+
+-}
+
+import Browser.Dom as Dom exposing (Element)
+import Html exposing (Html, div)
+import Html.Attributes as Attrs
+import Task exposing (Task)
+import Task.Extra as TaskExtra
 
 
 type Alignment
+    = Alignment Position Placement
+
+
+type Position
+    = Position { x : Float, y : Float, width : Float }
+
+
+type Placement
     = Above
     | Below
 
 
-init : { container : Element, select : Element } -> Alignment
-init { container, select } =
+type Params
+    = Params { container : Element, select : Element, viewport : Dom.Viewport }
```

Comment:
We can remove the `Params` type now.

Otherwise LGTM

Reply:
:checkered_flag: